dev/com.ibm.ws.security.authorization.jacc.web/test/com/ibm/ws/security/authorization/jacc/web/impl/WebJaccServiceImplTest.java

```diff
@@ -0,0 +1,395 @@
+        final String method = "GET";
+        final JaccServiceImpl jaccService = JaccServiceTestUtil.createJaccService(pcm);
+        context.checking(new Expectations() {
+            {
+                allowing(jaccProviderServiceRef).getProperty(JACC_POLICY_PROVIDER);
+                allowing(jaccProviderServiceRef).getProperty(JACC_POLICY_PROVIDER_EE9);
+                will(returnValue(JACC_POLICY_PROVIDER_IMPL));
+                allowing(jaccProviderServiceRef).getProperty(JACC_FACTORY);
+                allowing(jaccProviderServiceRef).getProperty(JACC_FACTORY_EE9);
+                will(returnValue(JACC_FACTORY_IMPL));
+                allowing(cc).locateService("jaccProviderServiceProxy", jaccProviderServiceProxyRef);
+                will(returnValue(jaccProviderServiceProxy));
+                allowing(cc).locateService("jaccProviderService", jaccProviderServiceRef);
+                will(returnValue(jaccProviderService));
+                allowing(cc).locateService("locationAdmin", wsLocationAdminRef);
+                will(returnValue(wsLocationAdmin));
+                allowing(jaccProviderServiceProxy).getPolicyProxy();
+                will(returnValue(policyProxy));
+                allowing(jaccProviderServiceProxy).getPolicyConfigFactory();
+                will(returnValue(pcf));
+                allowing(jaccProviderService).getPolicy();
+                will(returnValue(policy));
+                allowing(policyProxy).setPolicy();
+                allowing(policyProxy).refresh();
+                allowing(jaccProviderService).getPolicyConfigFactory();
+                will(returnValue(pcf));
+                allowing(jaccProviderServiceProxy).getPolicyName();
+                will(returnValue(JACC_POLICY_PROVIDER_IMPL));
+                allowing(jaccProviderServiceProxy).getFactoryName();
+                will(returnValue(JACC_FACTORY_IMPL));
+                allowing(wsLocationAdmin).resolveString("${wlp.user.dir}");
+                will(returnValue(directory));
+                allowing(wsLocationAdmin).getServerName();
+                will(returnValue(name));
+                allowing(cc).locateService("jaccService", jaccServiceRef);
+                will(returnValue(jaccService));
+                allowing(policyProxy).implies(with(any(String.class)), with(aNull(Subject.class)), with(any(WebUserDataPermission.class)));
+                will(returnValue(false));
+            }
+        });
+
+        try {
+            JaccServiceTestUtil.initJaccService(jaccService, jaccProviderServiceProxyRef, jaccProviderServiceRef, wsLocationAdminRef, cc);
+            webJaccService.setJaccService(jaccServiceRef);
+            webJaccService.activate(cc);
+            assertTrue(webJaccService.isSSLRequired(appName, moduleName, uriName, method, req));
+            // this is for null check
+            assertTrue(webJaccService.isSSLRequired(appName, moduleName, uriName, method, req));
```

Comment:
Not sure what the point of this second call is.

Reply:
This test is a move of the test from JaccServiceImplTest.java.  It says it is for a null check so I took it at its word.  The old isSSLRequiredInvalid test would pass a null value for the req so maybe that is what it was for.  I can delete the second one if you would like me to.  I was just keeping the tests the same as what they were.  I did go back to the creation of the test in 2015 and the code is the same since then for this particular part.

Reply:
I removed the second call.